src/topology/uniform_space/separation.lean

```diff
@@ -51,23 +113,65 @@ theorem separated_def' {α : Type u} [uniform_space α] :
 separated_def.trans $ forall_congr $ λ x, forall_congr $ λ y,
 by rw ← not_imp_not; simp [classical.not_forall]
 
+lemma id_rel_sub_separation_relation (α : Type*) [uniform_space α]: id_rel ⊆ 𝓢 α :=
+begin
+  unfold separation_rel,
+  rw id_rel_subset,
+  intros x,
+  suffices : ∀ t ∈ 𝓤 α, (x, x) ∈ t, by simpa only [refl_mem_uniformity],
+  exact λ t, refl_mem_uniformity,
+end
+
+lemma separation_rel_comap  {f : α → β} (h : ‹uniform_space α› = uniform_space.comap f ‹uniform_space β›) :
+𝓢 α = (prod.map f f) ⁻¹' 𝓢 β :=
+begin
+  dsimp [separation_rel],
+  rw [uniformity_comap h, (filter.comap_has_basis (prod.map f f) (𝓤 β)).sInter_sets,
+      ← preimage_bInter, sInter_eq_bInter],
+  refl,
+end
+
+protected lemma filter.has_basis.separation_rel {ι : Type*} {p : ι → Prop} {s : ι → set (α × α)}
+  (h : has_basis (𝓤 α) p s) :
+𝓢 α = ⋂ i ∈ set_of p, s i :=
+by { unfold separation_rel, rw h.sInter_sets }
+
+lemma separation_rel_eq_inter_closure : 𝓢 α = ⋂₀ (closure '' (𝓤 α).sets) :=
+by simpa [uniformity_has_basis_closure.separation_rel]
+
+lemma is_closed_separation_rel : is_closed (𝓢 α) :=
+begin
+  rw separation_rel_eq_inter_closure,
+  apply is_closed_sInter,
+  rintros _ ⟨t, t_in, rfl⟩,
+  exact is_closed_closure,
+end
+
 @[priority 100] -- see Note [lower instance priority]
 instance separated_t2 [s : separated α] : t2_space α :=
```

Comment:
I would remove the instance, and only keep it as a lemma. The reason is that there is a proof that `separated` implies `regular` below, and also an instance `regular -> t2`. So, this instance creates unnecessarily two paths from `separated` to `t2`. In fact, since you're giving the `iff` version below, you can even remove this completely!

Reply:
Yes, I agree. This is a historical instance, and the iff version was not there before.

Reply:
I tried to remove it, let's see what CI thinks about this.